tests/std/tests/P1132R7_out_ptr/test.cpp

```diff
@@ -0,0 +1,201 @@
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include <cassert>
+#include <memory>
+
+using namespace std;
+
+void test_raw_ptr() {
+    int i        = 5;
+    int* int_ptr = &i;
+
+    {
+        const auto f = [](int** ptr) {
+            assert(**ptr == 5);
+            static int i = 7;
+            *ptr         = &i;
+        };
+
+        f(inout_ptr(int_ptr));
+
+        assert(*int_ptr == 7);
+    }
+    {
+        const auto f = [](void** ptr) {
+            assert(*reinterpret_cast<int*>(*ptr) == 7);

```

Comment:
Suggestion: I believe this could be a `static_cast`. It's a stylistic choice whether to `static_cast` or `reinterpret_cast` from `void*` to `int*`; I believe that `static_cast` is preferable because this is reversing an implicit conversion (and providing type information). `reinterpret_cast` is "scarier" so I prefer to reserve it for when it's absolutely necessary. Occurs once below.

However, if you want to keep the `reinterpret_cast`s, that's fine.